docs/src/content/docs/get-started/realtime-leaderboard.mdx

```diff
@@ -0,0 +1,208 @@
+```
+
+Once the DiceDB server starts, you will see output similar to this
+
+```
+██████╗ ██╗ ██████╗███████╗██████╗ ██████╗
+██╔══██╗██║██╔════╝██╔════╝██╔══██╗██╔══██╗
+██║  ██║██║██║     █████╗  ██║  ██║██████╔╝
+██║  ██║██║██║     ██╔══╝  ██║  ██║██╔══██╗
+██████╔╝██║╚██████╗███████╗██████╔╝██████╔╝
+╚═════╝ ╚═╝ ╚═════╝╚══════╝╚═════╝ ╚═════╝
+
+2024-10-24T18:35:55Z INF starting DiceDB version=0.0.5
+2024-10-24T18:35:55Z INF running with port=7379
+2024-10-24T18:35:55Z INF running with enable-watch=true
+2024-10-24T18:35:55Z INF running with mode=multi-threaded num-shards=12
+2024-10-24T18:35:55Z INF ready to accept and serve requests on port=7379
+```
+
+### Starting the application server
+
+1. Clone the repository
+     ```bash
+     git clone https://github.com/arpitbbhayani/leaderboard-go-dicedb.git
+     cd leaderboard-go-dicedb
+     ```
+2. Start the application
+    ```bash
+    go run main.go
+    ```
+    This will start the application server on port 8080 by default, you should see output similar to
+    ```bash
+    2024/10/25 00:05:59 Server starting on :8080
+    ```
+
+### Interacting with the application
+1. Navigate to the application server from your desired browser at http://localhost:8080.
+2. Update player with their respective scores.
+3. As more player with scores get added, we can see players getting ranked accordingly.
+
+## Key Components
+
+1. DiceDB: As the in-memory data store to realtime track user scores.
+2. DiceDB Go SDK: To allow interaction between application server and DiceDB.
+3. `ZRANGEWATCH` command: Allows the application to subscribe to changes in the leaderboard.
+4. `ZADD` command: We'd leverage this command to add scores for users.
+5. Websocket: To push real-time updates to connected users.
+
+In this application, every player assigned with higher score is ranked at the top of leaderboard.
+
+```
+Client (WebSocket)  →  Go Application  →  DiceDB (ZADD)
+             ↑                                  ↓
+Leaderboard Update  ←  Go Application ← DiceDB (ZRANGEWATCH)
+                                         (Real-Time Update)
+```
+
+## Understanding Real-Time Reactivity with `ZRANGEWATCH`
+
+The `ZRANGEWATCH` command allows the application to subscribe to changes in the leaderboard.
+This allows users to eradicate the need to continuously poll the server as updates are automatically delivered whenever changes occur.
+`ZRANGEWATCH` only sends the relevant changes to clients, making the process highly efficient in terms of both bandwidth and processing power.
+
+### Flow of `ZRANGEWATCH`:
+
+1. Setup: The client subscribes to updates on the leaderboard using `ZRANGEWATCH`.
+2. Data Changes: Whenever a player's score is updated, DiceDB triggers an update.
+3. Push Notifications: The server pushes the updated scores to all connected clients through WebSocket.
+
+## [Code](https://github.com/arpitbbhayani/leaderboard-go-dicedb.git) overview
```

Comment:
Suggestion: we can provide the link at the end or the beginning. The hyperlink stands out from other heading. wdyt?
```suggestion
## Code overview
```
<img width="544" alt="image" src="https://github.com/user-attachments/assets/4d0bac79-fc41-4486-a386-426cd09d6970">

<img width="780" alt="image" src="https://github.com/user-attachments/assets/c3769e3b-55c2-42ef-a6b8-5ce53e9e3bbb">